### PR TITLE
Apps migration

### PR DESCRIPTION
**Pull request by rmweir**
**Problem:**
Customers want to be able to migrate apps.project to dashboard apps (apps.catalog). This is blocked by two issues:
Helm3 apps that were installed by creating an app.project may
have had some of their resources changed with kustomize. Some
of these changes were to matchLabel selectors which are
immutable. This caused any updates to a migrated app to
result in an error because the manifests that were
being applied would try to remove the matchLabels, which
cannot be changed.

Need to be able to delete outdated apps (apps.project) without uninstalling resources.

**Solution:**
Now, migrated apps will continue using
kustomize to make the same changes to the manifests as
before.

Add annotation that skips helm uninstall. Revisions will still
be deleted and app will be removed from namespace's app
annotation. This is instended to be used as part of the
upgrade process from apps.project to apps.catalog.

**Depends on(merge first):**
https://github.com/rancher/shell/pull/16

**Issues:**
https://github.com/rancher/rancher/issues/34700
https://github.com/rancher/rancher/issues/34701

### CODE REVIEW

**Review by jiaqiluo — DISMISSED** [review]
LGTM

**Review by StrongMonkey — DISMISSED** [review]

File: pkg/controllers/managementuserlegacy/helm/controller.go [hunk 1]
```diff
@@ -269,29 +269,31 @@ func (l *Lifecycle) Remove(obj *v3.App) (runtime.Object, error) {
...
-	}
-	defer os.RemoveAll(tempDirs.FullPath)
-	tokenName, err := l.writeKubeConfig(obj, tempDirs.KubeConfigFull, true)
-	if err != nil {
-		return obj, err
-	}
-	defer func() {
-		if err := l.systemTokens.DeleteToken(tokenName); err != nil {
-			logrus.Errorf("cleanup for helm token [%s] failed, will not retry: %v", tokenName, err)
+	if !(obj.Annotations["cattle.io/skipUninstall"] == "true") {
```
Comment (jiaqiluo) [inline pkg/controllers/managementuserlegacy/helm/controller.go:272]: A couple of  questions here:
1. In which case is this annotation used? In other words,  in which case that the uninstallation should be skipped?  
1. Is this annotation a new or existing one? 
2. Does UI add this annotation to the app?

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: pkg/controllers/managementuserlegacy/helm/controller.go (hunk 1 above)
Comment (rmweir) [reply pkg/controllers/managementuserlegacy/helm/controller.go:272]: This is used after an app has been fully migrated and a user would like to deleted the old norman app object. The annotation is new. No it is meant to be added manually for now.

Comment (jiaqiluo) [reply pkg/controllers/managementuserlegacy/helm/controller.go:272]: ok, it sounds like something we need to add to the Rancher document and release notes.

**Review by jiaqiluo — APPROVED** [review]
LGTM. 
Updates in the documentation and release notes are needed.

**Review by StrongMonkey — APPROVED** [review]

**Pull request merged**